psi4/src/psi4/libfock/jk.cc

```diff
@@ -71,10 +71,12 @@ JK::~JK() {}
 std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_ptr<BasisSet> auxiliary,
                                  Options& options, std::string jk_type) {
 
+    bool is_composite = jk_type.find("+") != std::string::npos; // does SCF_TYPE contain +?
+
     bool do_density_screen = options.get_str("SCREENING") == "DENSITY";
     bool do_df_scf_guess = options.get_bool("DF_SCF_GUESS");
     
-    bool can_do_density_screen = (jk_type == "DIRECT" || jk_type == "LINK");
+    bool can_do_density_screen = (jk_type == "DIRECT" || jk_type == "DIRECTDFJ+LINK");
```

Comment:
This is indeed intended behavior! Although it honestly might be better changed.

In the current version of the code, DFJCOSX fails the subsequent check using the can_do_density_screen variable, which happens when can_do_density_screen is set to false. This can be seen by looking at the original code before alteration here, where only the jk_types DIRECT and LINK result in can_do_density_screen=true, and not a jk_type of COSX. Practically, this means that SCF_TYPE=COSX throws an exception when SCREENING is set to DENSITY.  I believe the reason the reason things were programmed this way, is because SCREENING=DENSITY does density screening from within TwoBodyAOInt, while DFJCOSX has its own density screening scheme implemented within itself.

The intentional filter on both J+K here is a continuation of the current behavior, where using the COSX algorithm in conjunction with SCREENING=DENSITY throws an exception.